src/NSwag.CodeGeneration.CSharp/Templates/ClientTemplate.tt

```diff
@@ -86,7 +86,7 @@ public partial interface I<#=Model.Class#>
     }
 
 <#}#>
-    partial void PrepareRequest(System.Net.Http.HttpClient request, ref string url);
+    partial void PrepareRequest(System.Net.Http.HttpClient request, System.Text.StringBuilder urlBuilder);

```

Comment:
`PrepareRequest` implementations that are modifying the URL are one of the benefiters from the change.

I've made a change that brings the new and keeps the old in a compatible way.